easybuild/easyconfigs/f/fgbio/fgbio-2.2.1.eb

```
@@ -0,0 +1,42 @@
+easyblock = 'CmdCp'
+
+name = 'fgbio'
+version = '2.2.1'
+
+homepage = 'https://fulcrumgenomics.github.io/fgbio'
+description = "A set of tools to analyze genomic data with a focus on Next Generation Sequencing."
+
+toolchain = SYSTEM
+
+source_urls = ['https://github.com/fulcrumgenomics/fgbio/archive/']
+sources = ['%(version)s.zip']
+checksums = ['0810b10365f501bcba1e00e02bd4543270bc8eb8bfb18cca07e89bafedbc2a79']
+
+builddependencies = [
+    ('sbt', '1.3.13', '-Java-%(javaver)s'),
+]
+
+dependencies = [('Java', '1.8')]
```

unless there is a very strong reason not to, i think we should stop using the oracle java. 

If Java 8 is required, then
```suggestion
dependencies = [('Java', '8')]
```
or if possible
```suggestion
dependencies = [('Java', '11')]
```

---

According to their web page (https://github.com/fulcrumgenomics/fgbio): "Java SE 8 is required."
Regarding Oracle Java or any other version, I don't mind, I just copied the old EC and updated the software version and hash. BTW, I don't know how to distinguish Oracle Java from others in this EC.

---

you do it the way i describe in the suggested change above

---

Yes, but my subtle question remains: how can I distinguish Oracle Java from ACME Java, Microsoft Java and Trump Java?

---

Change done, BTW.

---

In terms of easyconfigs? Everything that auto-downloads the sources uses adoptium eclipse openjdk (GPLv2).
The old easyconfigs (Java 1.8) that require a manual download, and to sign away your first-born, are oracle.

---

Yes, in terms of EasyConfigs. Going to the sources is easy, but looking at an EasyConfig I cannot tell one from other. I don't know if `[('Java', '1.8')]` will require a manual download or if it will auto-download.
I think I signed away my first, second and third-born XD
And regarding this change... the switch from 1.8 to 8 broke the build.

---

Everything `Java-1.x` is oracle stuff, unless suffixed with `-OpenJDK`. I believe that all `Java` easyconfigs currently in this repo versions `8` and up use some flavour of OpenJDK. Most use the Adoptium builds, but I think there are a few older 11.x ones that don't.
Easyconfigs of the style `Java-<majorversion>.eb` are actually wrappers. They install and depend on the easyconfig of the latest minor release of that version (e.g. `Java-8.eb` currently builds `Java-8.362.eb`)

---

regarding the test suite failure, we'll need to introduce an easyconfig for `sbt` that uses `Java-8` instead of `Java-1.8`

---

I don't have that. I will take a look and try to add if it looks straight-forward, but I don't want to go into a chain of dependency adjustments.
BTW, given that Java version 1.8 (internal version number) and 8 (product, public-facing version number) are the same, changing from one to another should not make anything fail. But I understand that this knowledge cannot be automatically extracted by the code.

---

@Micket @jfgrimm are you happy with this one now?